File: api/Plutus.Infrastructure/Business/Transactions/TransactionListItem.cs
namespace Plutus.Infrastructure.Business.Transactions;

public class TransactionListItem
{
    public string Id { get; set; }
    public DateTime BookedAt { get; set; }
    public decimal Amount { get; set; }
    public TransactionType? Type { get; set; }
    public string ObligorId { get; set; }
    public string ObligorName { get; set; }
    public string CategoryId { get; set; }
    public string CategoryName { get; set; }
    public bool IsExcluded { get; set; }
    public bool IsSplitItem { get; set; }
    public bool IsFixedExpense { get; set; }
}

File: ui/src/common/dtos/TransactionListItem.ts
export interface TransactionListItem {
  id: string;
  bookedAt: string;
  amount: number;
  type: number;
  obligorId: string;
  obligorName: string;
  obligorDisplayName: string;
  categoryId: string;
  categoryName: string;
  isExcluded: boolean;
  isSplitItem: boolean;
  isFixedExpense: boolean;
}
